File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/NutritionOrder.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class NutritionOrder : DomainResource, IResource
{
    public required Base.ResourceReference Patient { get; set; }
    public NutritionOrder_OralDiet? OralDiet { get; set; }
    public string[]? InstantiatesCanonical { get; set; }
    public string[]? InstantiatesUri { get; set; }
    public string[]? Instantiates { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public required string DateTime { get; set; }
    public NutritionOrder_EnteralFormula? EnteralFormula { get; set; }
    public Base.CodeableConcept[]? FoodPreferenceModifier { get; set; }
    public required string Status { get; set; }
    public Base.CodeableConcept[]? ExcludeFoodModifier { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public required string Intent { get; set; }
    public Base.ResourceReference? Orderer { get; set; }
    public NutritionOrder_Supplement[]? Supplement { get; set; }
    public Base.ResourceReference[]? AllergyIntolerance { get; set; }

    public class NutritionOrder_OralDiet_Texture : BackboneElement
    {
        public Base.CodeableConcept? FoodType { get; set; }
        public Base.CodeableConcept? Modifier { get; set; }
    }

    public class NutritionOrder_OralDiet_Nutrient : BackboneElement
    {
        public Base.Quantity? Amount { get; set; }
        public Base.CodeableConcept? Modifier { get; set; }
    }

    public class NutritionOrder_OralDiet : BackboneElement
    {
        public Base.CodeableConcept[]? Type { get; set; }
        public NutritionOrder_OralDiet_Texture[]? Texture { get; set; }
        public NutritionOrder_OralDiet_Nutrient[]? Nutrient { get; set; }
        public string[]? Schedule { get; set; }
        public string? Instruction { get; set; }
        public Base.CodeableConcept[]? FluidConsistencyType { get; set; }
    }

    public class NutritionOrder_EnteralFormula_Administration : BackboneElement
    {
        public object? Rate    
        {
            get
            {
                if (RateRatio is not null)
                {
                    return RateRatio;
                }
        
                if (RateQuantity is not null)
                {
                    return RateQuantity;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.Ratio))
                {
                    RateRatio = (Base.Ratio)value;
                    return;
                }
        
                if (value?.GetType() == typeof(Base.Quantity))
                {
                    RateQuantity = (Base.Quantity)value;
                    return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.Quantity? Quantity { get; set; }
        public string? Schedule { get; set; }
        public Base.Ratio? RateRatio { get; set; }
        public Base.Quantity? RateQuantity { get; set; }
    }

    public class NutritionOrder_EnteralFormula : BackboneElement
    {
        public Base.CodeableConcept? AdditiveType { get; set; }
        public Base.Quantity? MaxVolumeToDeliver { get; set; }
        public Base.CodeableConcept? BaseFormulaType { get; set; }
        public Base.CodeableConcept? RouteofAdministration { get; set; }
        public string? AdditiveProductName { get; set; }
        public Base.Quantity? CaloricDensity { get; set; }
        public string? AdministrationInstruction { get; set; }
        public NutritionOrder_EnteralFormula_Administration[]? Administration { get; set; }
        public string? BaseFormulaProductName { get; set; }
    }

    public class NutritionOrder_Supplement : BackboneElement
    {
        public Base.CodeableConcept? Type { get; set; }
        public Base.Quantity? Quantity { get; set; }
        public string[]? Schedule { get; set; }
        public string? Instruction { get; set; }
        public string? ProductName { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/NutritionOrder.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import CodeableConcept
from base import Timing
from base import Quantity
from base import DomainResource
from base import Ratio
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class NutritionOrderOralDietTexture(BackboneElement):
    food_type: Optional[CodeableConcept] = None
    modifier: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class NutritionOrderOralDietNutrient(BackboneElement):
    amount: Optional[Quantity] = None
    modifier: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class NutritionOrderOralDiet(BackboneElement):
    type: Optional[List[CodeableConcept]] = field(default_factory=list)
    texture: Optional[List[NutritionOrderOralDietTexture]] = field(default_factory=list)
    nutrient: Optional[List[NutritionOrderOralDietNutrient]] = field(default_factory=list)
    schedule: Optional[List[Timing]] = field(default_factory=list)
    instruction: Optional[str] = None
    fluid_consistency_type: Optional[List[CodeableConcept]] = field(default_factory=list)

@dataclass(kw_only=True)
class NutritionOrderEnteralFormulaAdministration(BackboneElement):
    quantity: Optional[Quantity] = None
    schedule: Optional[Timing] = None
    rate_ratio: Optional[Ratio] = None
    rate_quantity: Optional[Quantity] = None

@dataclass(kw_only=True)
class NutritionOrderEnteralFormula(BackboneElement):
    additive_type: Optional[CodeableConcept] = None
    max_volume_to_deliver: Optional[Quantity] = None
    base_formula_type: Optional[CodeableConcept] = None
    routeof_administration: Optional[CodeableConcept] = None
    additive_product_name: Optional[str] = None
    caloric_density: Optional[Quantity] = None
    administration_instruction: Optional[str] = None
    administration: Optional[List[NutritionOrderEnteralFormulaAdministration]] = field(default_factory=list)
    base_formula_product_name: Optional[str] = None

@dataclass(kw_only=True)
class NutritionOrderSupplement(BackboneElement):
    type: Optional[CodeableConcept] = None
    quantity: Optional[Quantity] = None
    schedule: Optional[List[Timing]] = field(default_factory=list)
    instruction: Optional[str] = None
    product_name: Optional[str] = None

@dataclass(kw_only=True)
class NutritionOrder(DomainResource):
    patient: Reference
    oral_diet: Optional[NutritionOrderOralDiet] = None
    instantiates_canonical: Optional[List[str]] = field(default_factory=list)
    instantiates_uri: Optional[List[str]] = field(default_factory=list)
    instantiates: Optional[List[str]] = field(default_factory=list)
    encounter: Optional[Reference] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    date_time: str
    enteral_formula: Optional[NutritionOrderEnteralFormula] = None
    food_preference_modifier: Optional[List[CodeableConcept]] = field(default_factory=list)
    status: str
    exclude_food_modifier: Optional[List[CodeableConcept]] = field(default_factory=list)
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    intent: str
    orderer: Optional[Reference] = None
    supplement: Optional[List[NutritionOrderSupplement]] = field(default_factory=list)
    allergy_intolerance: Optional[List[Reference]] = field(default_factory=list)